linode/helper/compare.go

```diff
@@ -0,0 +1,61 @@
+package helper
+
+import (
+	"strings"
+	"time"
+)
+
+func CompareTimeStrings(t1, t2, timeFormat string) bool {
+	parsedT1, err := time.Parse(timeFormat, t1)
+	if err != nil {
+		return false
+	}
+
+	parsedT2, err := time.Parse(timeFormat, t2)
+	if err != nil {
+		return false
+	}
+
+	return parsedT1.Equal(parsedT2)
+}
+
+func CompareTimeWithTimeString(t1 *time.Time, t2 string, timeFormat string) bool {
+	parsedT2, err := time.Parse(timeFormat, t2)
+	if err != nil {
+		return false
+	}
+
+	return t1.Equal(parsedT2)
+}
+
+func StringListEqual(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	freq := make(map[string]int)
+	for _, v := range a {
+		freq[v]++
+	}
+	for _, v := range b {
+		freq[v]--
+		if freq[v] < 0 {
+			return false
+		}
+	}
+	return true
```

Comment:
This function intent to check whether the list contains the same elements regardless of order. And it can be used for checking whether the scopes was change semantically.

For example, the scope changed from `linodes:read_only lke:read_only` to `linodes:read_write lke:read_only` is a semantically change, but from `linodes:read_only lke:read_only` to `lke:read_only linodes:read_only` isn't.

I can put a comment there to say what it does compare. What do you think?

Reply:
Gotcha, I think renaming the function to `StringSetEqual` should be good enough to imply that order of the set doesn't matter 👍